execute/table.go

```diff
@@ -244,54 +228,36 @@ func AppendRecord(i int, cr flux.ColReader, builder TableBuilder) error {
 	return nil
 }
 
-// AppendMappedRecordWithDefaults appends the records from cr onto builder, using colMap as a map of builder index to cr index.
-// if an entry in the colMap indicates a mismatched column, a default value is assigned to the builder's column
-func AppendMappedRecordWithDefaults(i int, cr flux.ColReader, builder TableBuilder, colMap []int) error {
+// AppendMappedRecordWithNulls appends the records from cr onto builder, using colMap as a map of builder index to cr index.
+// if an entry in the colMap indicates a mismatched column, the column is created with null values
+// this function is null-safe
+func AppendMappedRecordWithNulls(i int, cr flux.ColReader, builder TableBuilder, colMap []int) error {
 	if len(colMap) != len(builder.Cols()) {
-		return errors.New("AppendMappedRecordWithDefaults: colMap must have an entry for each table builder column")
+		return errors.New("AppendMappedRecordWithNulls: colMap must have an entry for each table builder column")
 	}
-	// TODO(adam): these zero values should be set to null when we have null support
 	for j, c := range builder.Cols() {
 		var err error
-		switch c.Type {
-		case flux.TBool:
-			var val bool
-			if colMap[j] >= 0 {
-				val = cr.Bools(colMap[j]).Value(i)
-			}
-			err = builder.AppendBool(j, val)
-		case flux.TInt:
-			var val int64
-			if colMap[j] >= 0 {
-				val = cr.Ints(colMap[j]).Value(i)
-			}
-			err = builder.AppendInt(j, val)
-		case flux.TUInt:
-			var val uint64
-			if colMap[j] >= 0 {
-				val = cr.UInts(colMap[j]).Value(i)
-			}
-			err = builder.AppendUInt(j, val)
-		case flux.TFloat:
-			var val float64
-			if colMap[j] >= 0 {
-				val = cr.Floats(colMap[j]).Value(i)
-			}
-			err = builder.AppendFloat(j, val)
-		case flux.TString:
-			var val string
-			if colMap[j] >= 0 {
-				val = cr.Strings(colMap[j]).ValueString(i)
-			}
-			err = builder.AppendString(j, val)
-		case flux.TTime:
-			var val Time
-			if colMap[j] >= 0 {
-				val = values.Time(cr.Times(colMap[j]).Value(i))
+		var val values.Value
+		if colMap[j] >= 0 {
+			val = ValueForRow(cr, i, colMap[j])
+			err = builder.AppendValue(j, val)
+		} else {
+			switch c.Type {
```

Comment:
Is there a reason why `AppendNil` can't be used in this section?

Reply:
nope. I will fix it.